### DISCUSSION TITLE
Feature request: add JSON Lines output

### DISCUSSION THREAD

**Discussion by aborruso** [duckdb/duckdb#4226] [Ideas]
Hi,
I think it would be great to add [JSON Lines](https://jsonlines.org/), to duckdb output.
JSON Lines for a lot of task is much more comfortable.

Thank you

**Comment by lnkuiper** [comment]
Not sure what exactly you mean with this.

How would this work? Are you thinking about a new option for the COPY command or something else? Are there other systems that implement the desired functionality?

Our json implementation prints json in a line, and you can export that as csv to get line-delimited json output already.

**Reply by aborruso** [reply]
Hi @lnkuiper and thank you.

I'm thinking in example to the CLI, and to add the jsonl output.

```
duckdb -jsonl -c "select * from 'input.csv' where pop>100"
```

The [JSON Lines](https://jsonlines.org/) is not a JSON in a line, it is a convenient format for storing structured data that may be processed one record at a time, in which each Line is a Valid JSON.

Then it's not

```json
[{"name":"tom","age":42},{"name":"anna","age":22}]
```

it's

```json
{"name":"tom","age":42}
{"name":"anna","age":22}
```

A recent post about JSON lines https://dbconvert.com/blog/json-lines-data-stream/
